test/core/storage/trie/trie_db/trie_batch_test.cpp

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/trie/polkadot_trie_db/polkadot_trie_batch.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <testutil/storage/map_storage.hpp>
+#include "crypto/hasher/hasher_impl.hpp"
+#include "testutil/literals.hpp"
+#include "testutil/outcome.hpp"
+#include "testutil/storage/base_leveldb_test.hpp"
+#include "testutil/storage/polkadot_trie_db_printer.hpp"
+
+using namespace kagome::storage::trie;
+using kagome::storage::face::WriteBatch;
+using kagome::common::Buffer;
+using kagome::hash::HasherImpl;
+using testing::_;
+using testing::Invoke;
+using testing::Return;
+
+class TrieBatchTest : public test::BaseLevelDB_Test {
+ public:
+  TrieBatchTest() : BaseLevelDB_Test("/tmp/leveldbtest") {}
+
+  void SetUp() override {
+    open();
+    trie = std::make_unique<PolkadotTrieDb>(std::move(db_));
+  }
+
+  static const std::vector<std::pair<Buffer, Buffer>> data;
+
+  std::unique_ptr<PolkadotTrieDb> trie;
+};
+
+const std::vector<std::pair<Buffer, Buffer>> TrieBatchTest::data = {
```

Comment:
can it done be better with value-parameterized tests?

Reply:
Several data sets can be put to use with them, but this is done for the trie and is hardly needed for batch, as I think.

Reply:
could you please put it more clear?

Reply:
There are parametrised test for trie. It needs a lot of tests because it has very sophisticated inner logic. Whereas batch doesn't. So I don't think it needs parametrised tests, because they would likely tests more of Trie's logic, not the batch.